learning_journal_db/tests.py

```diff
@@ -0,0 +1,185 @@
+    engine = get_engine(settings)
+    Base.metadata.create_all(engine)
+
+    def teardown():
+        testing.tearDown()
+        transaction.abort()
+        Base.metadata.drop_all(engine)
+
+    request.addfinalizer(teardown)
+    return engine
+
+
+@pytest.fixture(scope='function')
+def new_session(sqlengine, request):
+    session_factory = get_session_factory(sqlengine)
+    dbsession = get_tm_session(session_factory, transaction.manager)
+
+    def teardown():
+        transaction.abort()
+
+    request.addfinalizer(teardown)
+    return dbsession
+
+# Testing the models.
+
+
+def test_model_gets_added(new_session):
+    """Test the creation of the new model."""
+    assert len(new_session.query(MyModel).all()) == 0
+    model = MyModel(title="test_day", body="test_body")
+    new_session.add(model)
+    new_session.flush()
+    assert len(new_session.query(MyModel).all()) == 1
+
+
+def test_model_gets_added_2(new_session):
+    """Test the creation of 2 new models in the row."""
+    assert len(new_session.query(MyModel).all()) == 0
+    model = MyModel(title="test_day", body="test_body")
+    new_session.add(model)
+    new_session.flush()
+    model = MyModel(title="test_day2", body="test_body2")
+    new_session.add(model)
+    new_session.flush()
+    assert len(new_session.query(MyModel).all()) == 2
+
+# Testing the views.
+
+
+def dummy_http_request(new_session):
+    test_request = testing.DummyRequest()
+    test_request.dbsession = new_session
+    return test_request
+
+
+def dummy_http_request_post(title, body, date, new_session):
+    """Define a dummy request with method="POST"."""
+    test_request = testing.DummyRequest()
+    test_request.dbsession = new_session
+    test_request.method = 'POST'
+    test_request.POST['title'] = title
+    test_request.POST['body'] = body
+    test_request.POST['date'] = date
+    return test_request
+
+
+def test_lists_title(new_session):
+    """Test whether lists() pull out correct info from db."""
+    from .views.default import lists
+    new_session.add(MyModel(title='test1', body='test2', date='test3'))
+    new_session.flush()
+    result = lists(dummy_http_request(new_session))
+    for entry in result['entries']:
+        assert entry.title == 'test1'
+
+
+def test_lists_body(new_session):
+    """Test whether lists() pull out correct info from db."""
+    from .views.default import lists
+    new_session.add(MyModel(title='test_1', body='test2', date='test3'))
+    new_session.flush()
+    result = lists(dummy_http_request(new_session))
+    for entry in result['entries']:
+        assert entry.body == 'test2'
+
+
+def test_lists_date(new_session):
+    """Test whether lists() pull out correct info from db."""
+    from .views.default import lists
+    new_session.add(MyModel(title='test1', body='test2', date='test3'))
+    new_session.flush()
+    result = lists(dummy_http_request(new_session))
+    for entry in result['entries']:
+        assert entry.date == 'test3'
```

Comment:
While this is great, it might be even better to see if there's actually a list object in the HTTP response.